datafusion/core/src/physical_plan/planner.rs

```diff
@@ -1544,20 +1540,22 @@ mod tests {
             .build()?;
         let execution_plan = plan(&logical_plan).await;
 
-        let expected_error = "Unsupported CAST from Utf8 to Boolean";
-        match execution_plan {
-            Ok(_) => panic!("Expected planning failure"),
-            Err(e) => assert!(
-                e.to_string().contains(expected_error),
-                "Error '{}' did not contain expected error '{}'",
-                e,
-                expected_error
-            ),
-        }
+        let e = execution_plan.unwrap_err().to_string();
```

Comment:
This is great -- arrow 16.0.0 contains support for casting `Utf8` to bool (https://github.com/apache/arrow-rs/pull/1738 - Thanks @MazterQyou) --  so I updated the test to try and cast `Utf8` to some struct.

I also took the opportunity to clean up the test in general